common/djangoapps/student/views.py

```diff
@@ -587,7 +589,11 @@ def dashboard(request):
     # Get any programs associated with courses being displayed.
     # This is passed along in the template context to allow rendering of
     # program-related information on the dashboard.
+    credential_configuration = CredentialsApiConfig.current()
```

Comment:
Keep this separate from the retrieval of course Programs. Interleaving it like this makes the code harder to read. Move it below, and add a comment explaining what's being done.